plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java

```diff
@@ -219,15 +219,21 @@ private void calculateMinimumGraphSize(final ElkNode elkgraph, final LGraph lgra
      */
     private void importFlatGraph(final ElkNode elkgraph, final LGraph lgraph) {
         // Transform the node's children, unless we're told not to
+        int index = 0;
         for (ElkNode child : elkgraph.getChildren()) {
             if (!child.getProperty(LayeredOptions.NO_LAYOUT)) {
+                child.setProperty(InternalProperties.MODEL_ORDER, index);
```

Comment:
To have it discussed: do we want to set this property always, or just if it is required? I'm fine with either, I guess.

Reply:
I guess it would be better to not set properties that are not needed. The property should only be set if the order should be preserved.
If one other processor might need/want something like this it would be a valid reason to always set it.

Reply:
In that case you still have to add a check whether order preserving is activated, right?

Reply:
Correct